pkg/controller/scylladbcluster/resource_test.go

```diff
@@ -2104,419 +1467,409 @@ func TestMakeScyllaDBDatacenters(t *testing.T) {
 
 				return cluster
 			}(),
-			remoteNamespaces: map[string]*corev1.Namespace{
-				"dc1-rkc": {
-					ObjectMeta: metav1.ObjectMeta{
-						Name: "scylla-aaa",
-					},
+			datacenter: dcFromSpec(0),
+			remoteNamespace: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "scylla-aaa",
 				},
 			},
-			remoteControllers: map[string]metav1.Object{
-				"dc1-rkc": &scyllav1alpha1.RemoteOwner{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "cluster-111",
-						Namespace: "scylla-aaa",
-						UID:       "1234",
-					},
+			remoteController: &scyllav1alpha1.RemoteOwner{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "cluster-111",
+					Namespace: "scylla-aaa",
+					UID:       "1234",
 				},
 			},
-			expectedScyllaDBDatacenters: map[string][]*scyllav1alpha1.ScyllaDBDatacenter{
-				"dc1-rkc": {
-					func() *scyllav1alpha1.ScyllaDBDatacenter {
-						dc := newBasicScyllaDBDatacenter("dc1", "scylla-aaa", []string{})
-						dc.Spec.RackTemplate.Placement = &scyllav1alpha1.Placement{
-							NodeAffinity: &corev1.NodeAffinity{
-								PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
-									{
-										Weight: 111,
-										Preference: corev1.NodeSelectorTerm{
-											MatchFields: []corev1.NodeSelectorRequirement{
-												{
-													Key:      "foo",
-													Operator: corev1.NodeSelectorOpIn,
-													Values:   []string{"foo"},
-												},
-											},
-											MatchExpressions: []corev1.NodeSelectorRequirement{
-												{
-													Key:      "foo",
-													Operator: corev1.NodeSelectorOpIn,
-													Values:   []string{"foo"},
-												},
-											},
+			expectedScyllaDBDatacenters: func() *scyllav1alpha1.ScyllaDBDatacenter {
```

Comment:
I'm struggling to review this file diff (`resource_test.go`) due to its size - most of which looks like reformatting/no-op refactor, but not all. Also, the commit message doesn't say much about what is changing here.

Can you help me understand what is changing in this test file?

Reply:
Thank you. The github diff makes it very difficult to review with the mechanical large-scale unit test change/refactor being part of a large change. No comments here, but avoiding this back and forth by extracting the "mechanical" change in some way in the future would be helpful.